src/Hedgehog.Xunit/Attributes.fs

```diff
@@ -100,3 +100,26 @@ type RecheckAttribute(recheckData) =
   let _recheckData : string = recheckData
 
   member internal _.GetRecheckData = _recheckData
+
+/// Set a Generator for a parameter of a test annotated with `Property`
+///
+/// Example usage:
+///
+/// ```
+///
+/// type ConstantInt(i: int) =
+///   inherit ParameterGeneraterBaseType<int>()
+///   override _.Generator = Gen.constant i
+///
+/// [<Property>]
+/// let ``is always 2`` ([<ConstantInt(2)>] i) =
+///   Assert.StrictEqual(2, i)
+///
+/// ```
+[<AbstractClass>]
+[<AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)>]
+type ParameterGeneraterBaseType<'a>() =
+  inherit Attribute()
+  
+  abstract member Generator : Gen<'a> 
```

Comment:
AFAIK Hedgehog doesn't really call anything `Generator`, so perhaps this could be called `Gen` too.

(My bias towards simple names, again...)

Reply:
Ok

Reply:
This gives me name collision, C# thinks I'm trying to recursively use Gen, so I've rolled it back you only use it once when constructing the type. 

```CSharp
public class Positive : ParameterGenerator<int>
{
                                                             ///Error Int32 not defined on Gen
  public override Gen<int> Gen => Gen.Int32(Range.Constant(1, Int32.MaxValue));
}
```